glib-macros/src/lib.rs

```diff
@@ -300,6 +301,34 @@ pub fn gboxed_derive(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
+/// Derive macro for defining a [`SharedType`]`::get_type` function and
+/// the [`glib::Value`] traits.
+///
+/// # Example
+///
+/// ```
+/// use glib::prelude::*;
+/// use glib::subclass::prelude::*;
+///
+/// #[derive(Clone, Debug, PartialEq, Eq)]
+/// struct MySharedInner {
+///   foo: String,
+/// }
+/// #[derive(Clone, Debug, PartialEq, Eq, glib::Shared)]
+/// #[shared(type_name = "MyShared")]
+/// struct MyShared(std::sync::Arc<MySharedInner>);
```

Comment:
I think a better way for this to work would be the require the inner struct to implement a trait, and then to implement this trait on `Arc<T>` and `Rc<T>`. That way it works for both, and users can define their own inner type by implementing a much simpler trait (only needs to provide ref/unref!) while still being able to make use of the derive macro here

Reply:
That makes a lot of sense indeed :) Thanks for the suggestion